src/test/java/seedu/address/testutil/MarkPersonDescriptorBuilder.java

```diff
@@ -0,0 +1,146 @@
+package seedu.address.testutil;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.MarkCommand.MarkPersonDescriptor;
+import seedu.address.model.person.Attendance;
+import seedu.address.model.person.GradeProgress;
+import seedu.address.model.person.Homework;
+import seedu.address.model.person.LessonPlan;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Session;
+import seedu.address.model.tag.Tag;
+
+/**
+ * A utility class to help with building MarkPersonDescriptor objects.
+ */
+public class MarkPersonDescriptorBuilder {
+
+    private MarkPersonDescriptor descriptor;
+
+    public MarkPersonDescriptorBuilder() {
+        descriptor = new MarkPersonDescriptor();
+    }
+
+    public MarkPersonDescriptorBuilder(MarkPersonDescriptor descriptor) {
+        this.descriptor = new MarkPersonDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditPersonDescriptor} with fields containing {@code person}'s details
```

Comment:
Should be `MarkPersonDescriptor` instead of Edit, same for the rest of the methods in this file

Reply:
resolved!